### PR TITLE
Palette PDE commands

### PR DESCRIPTION
**Pull request by karl-cardenas-coding**
## Describe the Change

This PR documents new Palette CLI subcommand introduced in Palette 4.1

[- Commands](https://deploy-preview-1725--docs-spectrocloud.netlify.app/palette-cli/commands/)

The page with new additions is the [pde page](https://deploy-preview-1725--docs-spectrocloud.netlify.app/palette-cli/commands/pde).

## Review Changes

💻 [Preview URL](https://deploy-preview-1725--docs-spectrocloud.netlify.app/palette-cli/commands)

🎫 [PPD-1382](https://spectrocloud.atlassian.net/browse/PPD-1382)

### CODE REVIEW

File: docs/docs-content/palette-cli/commands/pcg.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+---
+sidebar_label: "PCG"
+title: "PCG"
+description: "Reference resource for the pcg command."
+hide_table_of_contents: false
+sidebar_position: 30
+tags: ["palette-cli"]
+---
+
+The `pcg` subcommand supports Private Cloud Gateway (PCG) operations, such as installing a PCG cluster and validating its installation. A local [kind](https://kind.sigs.k8s.io/) cluster is created to facilitate creating the PCG cluster in the target environment. You do not need to install kind or any other dependencies, the CLI includes all the required dependencies to stand up the kind cluster.
```
Comment (ritawatson) [inline docs/docs-content/palette-cli/commands/pcg.md:10]:
```suggestion
The `pcg` command supports Private Cloud Gateway (PCG) operations, such as installing a PCG cluster and validating its installation. A local [kind](https://kind.sigs.k8s.io/) cluster is created to facilitate creating the PCG cluster in the target environment. You do not need to install kind or any other dependencies. The CLI includes all the required dependencies to set up the kind cluster.  
```

File: docs/docs-content/palette-cli/commands/pde.md [hunk 2]
```diff
@@ -0,0 +1,422 @@
+---
+sidebar_label: "PDE"
+title: "PDE"
+description: "Reference resource for the pde command."
+hide_table_of_contents: false
+sidebar_position: 40
+tags: ["palette-cli"]
+---
+
+The `pde` subcommand interacts with the Palette Dev Engine (PDE) platform and its resources. You can use the `pde` command to log in to Palette, manage virtual clusters, and switch the project scope.
```
Comment (ritawatson) [inline docs/docs-content/palette-cli/commands/pde.md:10]:
```suggestion
The `pde` command interacts with the Palette Dev Engine (PDE) platform and its resources. You can use the `pde` command to log in to Palette, manage virtual clusters, and switch the project scope. 
```

File: docs/docs-content/palette-cli/commands/pde.md [hunk 3]
```diff
@@ -0,0 +1,422 @@
...
+
+
+
+  ### Update
+
+  Use the `update` subcommand to update an App deployment. The `update` subcommand supports the following flags.
+
+  | **Flag**              | **Description**   | **Type**    |
+  | ------------------- | --------------------------------------------------------------------------|---------|
+  | `--name` | The name of the App deployment. Only Apps pending an App Profile update are eligble for an update. | string |
```
Comment (ritawatson) [inline docs/docs-content/palette-cli/commands/pde.md:100]:
```suggestion
  | `--name` | The name of the App deployment. Only Apps pending an App Profile update are eligible for an update. | string |
```

File: docs/docs-content/palette-cli/commands/pde.md [hunk 4]
```diff
@@ -0,0 +1,422 @@
...
+  | `--help` | Help for the `update` subcommand. | - |
+
+
+
+## App Profile
+
+Use the `app-profile` command to manage app profiles. The `app-profile` command supports the following subcommands.
```
Comment (ritawatson) [inline docs/docs-content/palette-cli/commands/pde.md:107]:
```suggestion
Use the `app-profile` subcommand to manage app profiles. The `app-profile` command supports the following subcommands.
```

File: docs/docs-content/palette-cli/commands/pde.md [hunk 5]
```diff
@@ -0,0 +1,422 @@
...
+
+  | **Flag**              | **Description**   | **Type**    |
+  |-------------------|-----------------------|---------|
+  | `--config-file` | The path to the app profile configuration file. | string |
+  | `--help` | Help for the `from-config` subcommand. | - |s
+
+
+:::tip
+
+You can create an app profile config file with the `interactive` command and using the `--config-only` flag.
```
Comment (ritawatson) [inline docs/docs-content/palette-cli/commands/pde.md:142]:
```suggestion
You can create an app profile config file by using the `interactive` subcommand with the `--config-only` flag.
```

File: docs/docs-content/palette-cli/commands/pde.md [hunk 6]
```diff
@@ -0,0 +1,422 @@
...
+  * `create` - Create a Palette Registry.
+
+  * `delete` - Delete a Palette Registry.
+
+  * `list`   - List Palette Registries. 
+
+
+  ### Create
+
+  The `create` subcommand creates a Palette Registry. You can specify an Open Container Initiative (OCI) or a Helm registry. Use the subcommand `oci` or `helm` to specify the type of registry you want to create. The `create` subcommand supports the following subcommands.
```
Comment (ritawatson) [inline docs/docs-content/palette-cli/commands/pde.md:241]: Does Open Container Initiative (OCI) need to modify anything here? Such as "...Open Container Initiative (OCI) _registry_ or a Helm registry."

Italics aren't intended for the doc - it's only intended to highlight the added word. :)

File: docs/docs-content/palette-cli/commands/pde.md [hunk 7]
```diff
@@ -0,0 +1,422 @@
...
+  ```shell hideClipboard
+  palette pde registry create oci --endpoint https://harbor.internal.abc.org  \
+  --name internal-packs --username harbor --password *********
+  ```
+
+
+
+  ### List
+
+  The `list` subcommand lists all Palette Registries availble in App Mode. The `list` subcommand supports the following flags.
```
Comment (ritawatson) [inline docs/docs-content/palette-cli/commands/pde.md:267]:
```suggestion
  The `list` subcommand lists all Palette Registries available in App Mode. The `list` subcommand supports the following flags.
```

File: docs/docs-content/palette-cli/commands/pde.md [hunk 8]
```diff
@@ -0,0 +1,422 @@
...
+
+  | **Flag**  | **Description**      | **Type**    |
+  |-----------|----------------------|---------|
+  | `--type` | The type of registry. Allowed values are `oci` and `helm`. | string |
+  | `--help` | Help for the `list` subcommand. | - |
+
+
+  ### Delete
+
+  The `delete` subcommand deletes a Palette Registry from App Mode. You can specify an Open Container Initiative (OCI) or a Helm registry. Use the type flag `--type` to specify the type of registry you want to delete. The `delete` subcommand supports the following flags.
```
Comment (ritawatson) [inline docs/docs-content/palette-cli/commands/pde.md:277]: "...specify an  Open Container Initiative (OCI) _registry_ or a Helm registry"?

File: docs/docs-content/palette-cli/commands/pde.md [hunk 9]
```diff
@@ -0,0 +1,422 @@
...
+  | `--name` | The name of the registry. | string |
+  | `--type` | The type of registry. Allowed values are `oci` and `helm`. | string |
+  | `--help` | Help for the `delete` subcommand. | - |
+
+## Virtual Cluster
+
+You can use the `virtual-cluster` subcommand to manage Palette Virtual Clusters. Below is a list of the supported subcommands you can use. Use the `--help` flag to learn more about each subcommand.
+
+- `create` - Create a Palette Virtual Cluster.
+
```
Comment (ritawatson) [inline docs/docs-content/palette-cli/commands/pde.md:290]: It seems occurrences that follow this initial "Palette Virtual Cluster" should be "virtual cluster" to follow our internal style.

File: docs/docs-content/palette-cli/commands/pde.md [hunk 10]
```diff
@@ -0,0 +1,422 @@
...
+- `list` - List Palette Virtual Clusters.
+
+
+- `resize` - Resize a Palette Virtual Cluster.
+
+
+### Create
+
+Use the `create` subcommand to create a Palette Virtual Cluster. The `create` subcommand supports the following flags.
```
Comment (ritawatson) [inline docs/docs-content/palette-cli/commands/pde.md:312]: Just "virtual cluster"?

File: docs/docs-content/palette-cli/commands/pde.md [hunk 11]
```diff
@@ -0,0 +1,422 @@
...
+
+| **Flag**              | **Description**                                                              | **Type**    |
+|-------------------|--------------------------------------------------------------------------|---------|
+| `--cluster-group-name` | The name of the Cluster Group. Defaults to the active cluster group.  | string |
+| `--cpu` | CPUs (optional). Defaults to 4 cores. (default 4) | int |
+| `--memory`| Memory (GB) (optional). Defaults to 4GB. (default 4) | int|
+| `--name` | The name of the Virtual Cluster. | string |
+| `--storage`|  Storage (GB) (optional). Defaults to 4GB. (default 4)| int |
+| `--tags`|  A list of tags seperated by commas  (optional) | strings |
```
Comment (ritawatson) [inline docs/docs-content/palette-cli/commands/pde.md:321]:
```suggestion
| `--tags`|  A list of tags separated by commas  (optional) | strings |
```

**Review by ritawatson — APPROVED** [review]
Approving with a few items to address. 
Moving commands to the nav pane makes it so much easier to locate and navigate the commands - much better user experience.

We need to be clear about command vs. subcommand. I noted where I saw discrepancies. The metadescriptions look correct. A few typos and style things to address.

File: docs/docs-content/palette-cli/commands/ec.md [hunk 12]
```diff
@@ -0,0 +1,152 @@
+---
+sidebar_label: "EC"
+title: "EC"
+description: "Reference resource for the EC command."
+hide_table_of_contents: false
+sidebar_position: 10
+tags: ["palette-cli"]
+---
+
+The `ec` subcommand installs a self-hosted Palette Enterprise Cluster (EC) in your target environment. The installation is conducted through an interactive wizard that guides you through the various install configurations available. A local kind cluster is created to facilitate creating the Enterprise cluster in the target environment. You do not need to install kind or any other dependencies. The CLI includes all the required dependencies to stand up the kind cluster. You can use the `ec` command to install a [self-hosted Palette](../../enterprise-version/install-palette/install-palette.md) instance or a self-hosted [VerteX](../../vertex/install-palette-vertex/install-palette-vertex.md) instance.
```
[the remark this answers is not in the document]
Comment (karl-cardenas-coding) [reply docs/docs-content/palette-cli/commands/ec.md:10]: Nice catch @ritawatson

File: docs/docs-content/palette-cli/commands/pde.md (hunk 6 above)
Comment (karl-cardenas-coding) [inline docs/docs-content/palette-cli/commands/pde.md:241]:
```suggestion
  The `create` subcommand creates a Palette Registry. You can specify an Open Container Initiative (OCI) registry or a Helm registry. Use the subcommand `oci` or `helm` to specify the type of registry you want to create. The `create` subcommand supports the following subcommands. 
```

File: docs/docs-content/palette-cli/commands/pde.md (hunk 8 above)
Comment (karl-cardenas-coding) [inline docs/docs-content/palette-cli/commands/pde.md:277]:
```suggestion
  The `delete` subcommand deletes a Palette Registry from App Mode. You can specify an Open Container Initiative (OCI) registry or a Helm registry. Use the type flag `--type` to specify the type of registry you want to delete. The `delete` subcommand supports the following flags.
```

File: docs/docs-content/palette-cli/commands/pde.md [hunk 13]
```diff
@@ -0,0 +1,422 @@
...
+  |-------------------|--------------------------------------------------------------------------|---------|
+  | `--name` | The name of the registry. | string |
+  | `--type` | The type of registry. Allowed values are `oci` and `helm`. | string |
+  | `--help` | Help for the `delete` subcommand. | - |
+
+## Virtual Cluster
+
+You can use the `virtual-cluster` subcommand to manage Palette Virtual Clusters. Below is a list of the supported subcommands you can use. Use the `--help` flag to learn more about each subcommand.
+
+- `create` - Create a Palette Virtual Cluster.
```
Comment (karl-cardenas-coding) [inline docs/docs-content/palette-cli/commands/pde.md:289]:
```suggestion
- `create` - Create a virtual cluster.
```

File: docs/docs-content/palette-cli/commands/pde.md (hunk 10 above)
Comment (karl-cardenas-coding) [inline docs/docs-content/palette-cli/commands/pde.md:312]:
```suggestion
Use the `create` subcommand to create a virtual cluster. The `create` subcommand supports the following flags.
```

**Pull request merged**